06_puppy/jimbotech/puppy.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	var puppyStore Storer = NewPuppyStorer()
+	pup := Puppy{1, "kelpie", "brown", "indispensable"}
```

Comment:
`indispensable` is not a colour, and the evidence I saw showed that the brown kelpie was not indispensable.

More seriously, the order of these parameters does not match the order declared in the struct. I suggest you use named parameters:
```
pup := Puppy{Breed: "kelpie", Colour: "brown", breed: "indispensable"}
```
That means you are now correct, and you get to leave out `ID`, which it does not make sense to put it because you are not meant to set that.

Since you've made this mistake throughout your code, you should be asking why your tests did not catch this, and how you can update your tests to catch it.

hint:
```
s.Equal("brown", p.Colour)
...
```
instead of
```
s.Equal(pup1, pup2)
```

Reply:
I think you were getting tired ... there is a lot of breeding going on here ... apart from that the order was correct ... but it is a much better idea to use name parameters as you suggest, so you win again. Hurray!